### PR TITLE
chore: Upgrade golangci-lint to 1.47

### PR DESCRIPTION
**Pull request by kakkoyun**
Enable disabled linters

Fix issues

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

**Review by metalmatze — APPROVED** [review]

File: pkg/metastore/kv.go [hunk 1]
```diff
@@ -85,10 +85,18 @@ func MakeLocationID(l *pb.Location) string {
 		// 0. This works out well as the key is byte aligned to the nearest 8
 		// bytes that way.
 
-		//nolint:errcheck // ignore error as writing to the hash will cannot error
+		//nolint:errcheck,staticcheck
+		// ignore error as writing to the hash will cannot error
```
Comment (metalmatze) [inline pkg/metastore/kv.go:89]:
```suggestion
		// ignore the error as writing to the hash will not error
```

File: pkg/metastore/kv.go [hunk 2]
```diff
@@ -85,10 +85,18 @@ func MakeLocationID(l *pb.Location) string {
...
+		// https://staticcheck.io/docs/checks#SA1003
+		// The encoding/binary package can only serialize types with known sizes.
+		// This precludes the use of the int and uint types, as their sizes differ on different architectures.
+		// TODO: Fix this.
 		binary.Write(hash, binary.BigEndian, 1)
 	} else {
-		//nolint:errcheck // ignore error as writing to the hash will cannot error
+		//nolint:errcheck,staticcheck
+		// ignore error as writing to the hash will cannot error
```
Comment (kakkoyun) [inline pkg/metastore/kv.go:97]:
```suggestion
		// ignore the error as writing to the hash will not error
```

File: pkg/metastore/kv.go [hunk 3]
```diff
@@ -85,10 +85,18 @@ func MakeLocationID(l *pb.Location) string {
 		// 0. This works out well as the key is byte aligned to the nearest 8
 		// bytes that way.
 
-		//nolint:errcheck // ignore error as writing to the hash will cannot error
+		//nolint:errcheck,staticcheck
+		// ignore the error as writing to the hash will not error
+		// https://staticcheck.io/docs/checks#SA1003
+		// The encoding/binary package can only serialize types with known sizes.
+		// This precludes the use of the int and uint types, as their sizes differ on different architectures.
+		// TODO: Fix this.
```
Comment (brancz) [inline pkg/metastore/kv.go:93]: I don't understand what needs to be fixed here.`hash` is fixed size, so the staticcheck link doesn't apply here.

Comment (kakkoyun) [reply pkg/metastore/kv.go:93]: If I don't understand this incorrectly, `int` is arch dependent so var sized and staticcheck wants us to specify it.

Comment (brancz) [reply pkg/metastore/kv.go:93]: Ahhhhhh I see now. Then we need to change this to `uint64(0)` and `uint64(1)`.

**Comment by kakkoyun** [comment]
:see_no_evil: I've forgotten that I've marked this as auto-merge. I'll send a fix for the issue as a follow up

**Pull request merged**